src/btree/bt_handle.c

```diff
@@ -437,6 +437,14 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
 		FLD_SET(btree->assert_flags, WT_ASSERT_COMMIT_TS_KEYS);
 	else if (WT_STRING_MATCH("never", cval.str, cval.len))
 		FLD_SET(btree->assert_flags, WT_ASSERT_COMMIT_TS_NEVER);
+	WT_RET(__wt_config_gets(session,
+	    cfg, "assert.durable_timestamp", &cval));
+	/* Durable timestamp make sense only with a commit timestamp. */
+	if (WT_STRING_MATCH("always", cval.str, cval.len))
+		FLD_SET(btree->assert_flags,
+		    WT_ASSERT_DURABLE_TS_ALWAYS | WT_ASSERT_COMMIT_TS_ALWAYS);
+	else if (WT_STRING_MATCH("never", cval.str, cval.len))
+		FLD_SET(btree->assert_flags, WT_ASSERT_DURABLE_TS_NEVER);
```

Comment:
Should this contain `WT_ASSERT_COMMIT_TS_NEVER`? That is, are durable and commit timestamps always together?

Reply:
There can be a commit timestamp without a durable timestamp, but vice versa should never happen. Durable timestamp is only applicable for prepared transactions for which commit timestamp is mandatory. I will modify the `WT_ASSERT_DURABLE_TS_ALWAYS` to include `WT_ASSERT_COMMIT_TS_ALWAYS` .